pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -309,18 +377,18 @@ func (r *ReconcileOneAgent) updateCR(instance *dynatracev1alpha1.OneAgent) error
 
 	newStatus := instance.Status
 
-	// Rather than sending the existing value, the .status.items map also gets replaced in-place, so we send a
-	// dummy object to avoid modifying it.
-	instance.Status = dynatracev1alpha1.OneAgentStatus{}
-
-	if err := r.client.Update(context.TODO(), instance); err != nil {
+	instance.Status = newStatus
```

Comment:
You don't really need this instruction. Or at least run it after `.Update()`. but don't think it would be used.